### ISSUE TITLE
Fix `defer` attribute is incorrectly required on `script[type="text/babel"]` elements

### ISSUE THREAD

**Issue by reosablo** [markuplint/markuplint#1334]
- Markuplint version: `3.15.0`
- Parser lang: HTML
- Node.js version: `20.9.0`
- OS: Windows

## Describe the bug

Markuplint incorrectly requires `defer` attribute on `script[type="text/babel"]` elements used by [`@babel/standalone`](https://babeljs.io/docs/babel-standalone) even if `[data-type="module"]` attribute is specified on them.

## Code Example

```html:index.html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Document</title>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js" defer></script>
  <script type="text/babel" data-type="module" src="main.js"></script>
</head>
<body>
  <h1>Document</h1>
</body>
</html>
```

## Steps To Reproduce

1. `npx markuplint index.html`

## The current behavior

```console
$ npx markuplint index.html
<markuplint> error: The "script" element expects the "defer" attribute (required-attr) /path/to/index.html:7:3
  6: ••<script•src="https://unpkg.com/@babel/standalone/babel.min.js"•defer></script>
  7: ••<script type="text/babel" data-type="module" src="main.js"></script>
  8: </head>
```

## The expected behavior

```console
$ npx markuplint index.html
<markuplint> passed /path/to/index.html
```

## Related Issues

- #438
- #439

**Comment by reosablo** [comment]
Ideally, `defer` attribute should be required only for [classic scripts](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element) but the selector in [`preset.performance.json`](https://github.com/markuplint/markuplint/blob/6fd54a514ecd78cb2d75370616883fd7ad81f5e4/packages/%40markuplint/config-presets/src/preset.performance.json#L20) will become too complex for maintainers. see [JavaScript MIME type](https://mimesniff.spec.whatwg.org/#javascript-mime-type).

```jsonc
{
	/**
	 * Require `defer` attr
	 *
	 * Should load and parse scripts lazily to avoid render-blocking.
	 */
	"selector": ":where(script[src]:is(:not([type]),[type=\"application/ecmascript\" i],[type=\"application/javascript\" i],[type=\"application/x-ecmascript\" i],[type=\"application/x-javascript\" i],[type=\"text/ecmascript\" i],[type=\"text/javascript\" i],[type=\"text/javascript1.0\" i],[type=\"text/javascript1.1\" i],[type=\"text/javascript1.2\" i],[type=\"text/javascript1.3\" i],[type=\"text/javascript1.4\" i],[type=\"text/javascript1.5\" i],[type=\"text/jscript\" i],[type=\"text/livescript\" i],[type=\"text/x-ecmascript\" i],[type=\"text/x-javascript\" i]))",
	"rules": {
		"required-attr": ["defer"]
	}
}
```

The selector matches the flowing elements: 

- `script[src]:not([type])`
- `script[src][type="application/ecmascript" i]`
- `script[src][type="application/javascript" i]`
- `script[src][type="application/x-ecmascript" i]`
- `script[src][type="application/x-javascript" i]`
- `script[src][type="text/ecmascript" i]`
- `script[src][type="text/javascript" i]`
- `script[src][type="text/javascript1.0" i]`
- `script[src][type="text/javascript1.1" i]`
- `script[src][type="text/javascript1.2" i]`
- `script[src][type="text/javascript1.3" i]`
- `script[src][type="text/javascript1.4" i]`
- `script[src][type="text/javascript1.5" i]`
- `script[src][type="text/jscript" i]`
- `script[src][type="text/livescript" i]`
- `script[src][type="text/x-ecmascript" i]`
- `script[src][type="text/x-javascript" i]`

**Comment by YusukeHirao** [comment]
@reosablo 

Thank you for bringing this issue to our attention. I understand your concern regarding the defer attribute requirement for script[type="text/babel"] elements, primarily when used with @babel/standalone.

Markuplint aims to adhere to HTML standards, and @babel/standalone falls slightly outside these specifications. Therefore, Markuplint treats these cases according to the general HTML guidelines, which might not align perfectly with specific JavaScript tools or libraries.

In this scenario, I recommend customizing your Markuplint configuration to better suit your project's needs. You can adjust the rules to ignore attributes or elements specific to @babel/standalone. This way, you can maintain the integrity of your project's code while leveraging the benefits of Markuplint.

Please feel free to reach out if you need further assistance configuring Markuplint for your specific use case. We're here to help ensure that your experience with Markuplint is as smooth and efficient as possible.
